src/util.test.ts

```diff
@@ -107,6 +107,73 @@ describe('util', () => {
     });
   });
 
+  describe('gweiDecToWEIBN', () => {
+    it('should convert a whole number to WEI', () => {
+      expect(util.gweiDecToWEIBN(1).toNumber()).toBe(1000000000);
+      expect(util.gweiDecToWEIBN(123).toNumber()).toBe(123000000000);
+      expect(util.gweiDecToWEIBN(101).toNumber()).toBe(101000000000);
+      expect(util.gweiDecToWEIBN(1234).toNumber()).toBe(1234000000000);
+    });
+
+    it('should convert a number with a decimal part to WEI', () => {
+      expect(util.gweiDecToWEIBN(1.1).toNumber()).toBe(1100000000);
+      expect(util.gweiDecToWEIBN(123.01).toNumber()).toBe(123010000000);
+      expect(util.gweiDecToWEIBN(101.001).toNumber()).toBe(101001000000);
+      expect(util.gweiDecToWEIBN(1234.567).toNumber()).toBe(1234567000000);
+    });
+
+    it('should convert a number < 1 to WEI', () => {
+      expect(util.gweiDecToWEIBN(0.1).toNumber()).toBe(100000000);
+      expect(util.gweiDecToWEIBN(0.01).toNumber()).toBe(10000000);
+      expect(util.gweiDecToWEIBN(0.001).toNumber()).toBe(1000000);
+      expect(util.gweiDecToWEIBN(0.567).toNumber()).toBe(567000000);
+    });
+
+    it('should round to whole WEI numbers', () => {
+      expect(util.gweiDecToWEIBN(0.1001).toNumber()).toBe(100100000);
+      expect(util.gweiDecToWEIBN(0.0109).toNumber()).toBe(10900000);
+      expect(util.gweiDecToWEIBN(0.0014).toNumber()).toBe(1400000);
+      expect(util.gweiDecToWEIBN(0.5676).toNumber()).toBe(567600000);
+    });
+
+    it('should handle NaN', () => {
+      expect(util.gweiDecToWEIBN(NaN).toNumber()).toBe(0);
+    });
+  });
+
+  describe('weiHexToGweiDec', () => {
```

Comment:
Might be personal preference, but I think these tests can be clearer in intent, and also less prone to error, if the hard coded inputs and hard coded outputs are explicitly provided, as opposed to be calculated by other functions.

This is how I would write these tests, feel free to copy and paste:

```
describe('weiHexToGweiDec', () => {
  it('should convert a whole number to WEI', () => {
    const testData = [
      {
        input: '3b9aca00',
        expectedResult: 1,
      },
      {
        input: '1ca35f0e00',
        expectedResult: 123,
      },
      {
        input: '178411b200',
        expectedResult: 101,
      },
      {
        input: '11f5021b400',
        expectedResult: 1234,
      },
    ];
    testData.forEach(({ input, expectedResult }) => {
      expect(
        util.weiHexToGweiDec(input),
      ).toBe(expectedResult);
    });
  });

  it('should convert a number with a decimal part to WEI', () => {
    const testData = [
      {
        input: '4190ab00',
        expectedResult: 1.1,
      },
      {
        input: '1ca3f7a480',
        expectedResult: 123.01,
      },
      {
        input: '178420f440',
        expectedResult: 101.001,
      },
      {
        input: '11f71ed6fc0',
        expectedResult: 1234.567,
      },
    ];

    testData.forEach(({ input, expectedResult }) => {
      expect(
        util.weiHexToGweiDec(input),
      ).toBe(expectedResult);
    });
  });

  it('should convert a number < 1 to WEI', () => {
    const testData = [
      {
        input: '5f5e100',
        expectedResult: 0.1,
      },
      {
        input: '989680',
        expectedResult: 0.01,
      },
      {
        input: 'f4240',
        expectedResult: 0.001,
      },
      {
        input: '21cbbbc0',
        expectedResult: 0.567,
      },
    ];

    testData.forEach(({ input, expectedResult }) => {
      expect(
        util.weiHexToGweiDec(input),
      ).toBe(expectedResult);
    });
  });

  it('should work with 0x prefixed values', () => {
    expect(util.weiHexToGweiDec('0x5f48b0f7')).toBe('1.598599415');
  });
});
```

Reply:
I was not going to comment about this, but jest has a [`test.each` function](https://jestjs.io/docs/api#testeachtablename-fn-timeout) which is my personal preference, either as string literal table or array table.

Example:
```ts
test.each`
  a    | b    | expected
  ${1} | ${1} | ${2}
  ${1} | ${2} | ${3}
  ${2} | ${1} | ${3}
`('returns $expected when $a is added $b', ({a, b, expected}) => {
  expect(a + b).toBe(expected);
});
```

Example from mobile:
https://github.com/MetaMask/metamask-mobile/blob/978b73d88a5208a7d5d74415af980a2f3a59403d/app/components/Base/Keypad/createKeypadRule.test.js

Reply:
copy and pasted, just had to update the expectedResult to strings

Reply:
3a42d98